### ISSUE TITLE
[no-untyped-public-signature] Requires return type on setters, which cannot have a return type

### ISSUE THREAD

**Issue by glen-84** [typescript-eslint/typescript-eslint#1263]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-untyped-public-signature": "error"
  }
}
```

```TS
class Example {
    private _x: string;

    public get x(): string {
        return this._x;
    }

    public set x(x: string) {
        this._x = x;
    }
}
```

**Expected Result**

No errors.

**Actual Result**

Error: `Public method has no return type`

**Additional Info**

Related to #1229.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.9.0` |
| `@typescript-eslint/parser`        | `2.9.0` |
| `TypeScript`                       | `3.4.5` |
| `ESLint`                           | `6.7.1` |
| `node`                             | `12.13.1` |
| `npm`                              | `6.12.1` |
